src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+        wrappingU = toWrapping(mapMode);
+    }
+    if(mat->Get(AI_MATKEY_MAPPINGMODE_V(type, 0), mapMode) == AI_SUCCESS) {
+        wrappingV = toWrapping(mapMode);
+    }
+
+    return TextureData{TextureData::Type::Texture2D, filterU, filterV, {Sampler::Mipmap::Linear}, {wrappingU, wrappingV, wrappingW}, id, &_textures[id]};
+}
+
+UnsignedInt AssimpImporter::doImage2DCount() const { return _textures.size(); }
+
+std::optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id) {
+    const aiMaterial* mat;
+    aiTextureType type;
+    std::tie(mat, type) = _textures[id];
+
+    aiString texturePath;
+    if(mat->Get(AI_MATKEY_TEXTURE(type, 0), texturePath) != AI_SUCCESS) {
+        Error() << "Trade::AssimpImporter::image2D(): Error getting path for texture with id" << id;
+        return std::nullopt;
+    }
+
+    std::string path = texturePath.C_Str();
+    if(path[0] == '*') {
+        /* Load embedded texture */
+        char* err;
+        const Int index = Int(strtol(path.substr(0).c_str(), &err, 10));
+        if(err == nullptr) {
+            Error() << "Trade::AssimpImporter::image2D(): Embedded texture path did not contain a valid integer string.";
+            return std::nullopt;
+        }
+        const aiTexture* texture = _scene->mTextures[index];
+
+        if(texture->mHeight == 0) {
+            /* Compressed image data */
+            auto textureData = Containers::ArrayView<const char>(reinterpret_cast<const char*>(texture->pcData), texture->mWidth);
+
+            std::string importerName;
+            if(texture->CheckFormat("dds")) {
+                importerName = "DdsImporter";
+            } else if(texture->CheckFormat("jpg")) {
+                importerName = "JpegImporter";
+            } else if(texture->CheckFormat("pcx")) {
+                importerName = "PxcImporter";
+            } else if(texture->CheckFormat("png") || std::strncmp(textureData.suffix(1).data(), "PNG", 3) == 0) {
+                importerName = "PngImporter";
+            } else {
+                Error() << "Trade::AssimpImporter::image2D(): Could not detect filetype of embedded data.";
+                return std::nullopt;
+            }
+
+            std::unique_ptr<Trade::AbstractImporter> importer =
+                static_cast<PluginManager::Manager<Trade::AbstractImporter>&>(*manager()).loadAndInstantiate(importerName);
+            if(!importer) {
+                Error() << "Trade::AssimpImporter::image2D(): Could not find importer for embedded data.";
+                return std::nullopt;
+            }
+            importer->openData(textureData);
+            return importer->image2D(0);
+        } else {
+            /* Uncompressed image data */
+            const Vector2i dimensions{Int(texture->mHeight), Int(texture->mWidth)};
+            size_t size = dimensions.product()*4;
+            Containers::Array<char> data{Containers::NoInit, size};
+            std::memcpy(texture->pcData, data.data(), size);
```

Comment:
Hah, it's really confusing: `unsigned char b,g,r,a;` is the definition in the code.
Not even sure if it is okay to assume any order here. There is a function to convert to `aiColor4D`, which I will use in this case.